ste/xfer-deleteBlob.go

```diff
@@ -60,7 +61,14 @@ func doDeleteBlob(jptm IJobPartTransferMgr, p pipeline.Pipeline) {
 
 	// note: if deleteSnapshotsOption is 'only', which means deleting all the snapshots but keep the root blob
 	// we still count this delete operation as successful since we accomplished the desired outcome
-	_, err := srcBlobURL.Delete(jptm.Context(), jptm.DeleteSnapshotsOption().ToDeleteSnapshotsOptionType(), azblob.BlobAccessConditions{})
+	err := error(nil)
+	if jptm.PermanentDeleteOption().ToPermanentDeleteOptionType() == azblob.BlobDeletePermanent {
+		// Change service version before performing permanent delete
+		ctx := context.WithValue(jptm.Context(), ServiceAPIVersionOverride, "2020-08-04")
```

Comment:
We've already updated AZCOPY_DEFAULT_SERVICE_API_VERSION=2020-08-04. So we can remove the hardcoded string.